mycroft/skills/media/__init__.py

```diff
@@ -0,0 +1,130 @@
+        config = ConfigurationManager.get()
+        self.base_conf = config.get('Media')
+
+    def initialize(self):
+        logger.info('Initializing MediaSkill commons')
+        logger.info('loading vocab files from ' + join(dirname(__file__),
+                    'vocab', self.lang))
+        self.load_vocab_files(join(dirname(__file__), 'vocab', self.lang))
+
+        self.register_vocabulary(self.name, 'NameKeyword')
+        self._register_common_intents()
+        self._register_event_handlers()
+
+    def _register_common_intents(self):
+        """
+           Register common intents, these include basically all intents
+           except the intents to start playback.
+        """
+        intent = IntentBuilder('NextIntent').require('NextKeyword')
+        self.register_intent(intent, self.handle_next)
+
+        intent = IntentBuilder('PrevIntent').require('PrevKeyword')
+        self.register_intent(intent, self.handle_prev)
+
+        intent = IntentBuilder('PauseIntent').require('PauseKeyword')
+        self.register_intent(intent, self.handle_pause)
+
+        intent = IntentBuilder('PlayIntent') \
+            .one_of('PlayKeyword', 'ResumeKeyword')
+        self.register_intent(intent, self.handle_play)
+
+        intent = IntentBuilder('CurrentlyPlayingIntent')\
+            .require('CurrentlyPlayingKeyword')
+        self.register_intent(intent, self.handle_currently_playing)
+
+    def _register_event_handlers(self):
+        """
+           Register event handlers for stopping currently playing media
+           when new media is started and handlers for lowering media volume
+           while mycroft is speaking.
+        """
+        self.emitter.on('mycroft.media.stop', self.handle_stop)
+        self.emitter.on('recognizer_loop:audio_output_start',
+                        self.lower_volume)
+        self.emitter.on('recognizer_loop:audio_output_end',
+                        self.restore_volume)
+
+    def handle_next(self, message):
+        """
+           handle_next() should be implemented by the skill to switch to next
+           song/channel/video in queue.
+        """
+        logger.debug('handle_next not implemented in ' + self.name)
+
+    def handle_prev(self, message):
+        """
+           handle_prev() should be implemented by the skill to switch to
+           previous song/channel/video in queue
+        """
+        logger.debug('handle_prev not implemented in ' + self.name)
+
+    def handle_currently_playing(self, message):
+        """
+           handle_currently_playing() should be implemented to tell the user
+           what is currently playing
+        """
+        logger.debug('handle_currently_playing not implemented in ' +
+                     self.name)
+
+    def play(self):
+        """ Stop currently playing media before starting the new. """
+        logger.info('Stopping currently playing media if any')
+        self.emitter.emit(Message('mycroft.media.stop'))
```

Comment:
Maybe we could add some data along `media.stop` to identify the sender. Something like:
`self.emitter.emit(Message('mycroft.media.stop', metadata={sender=self.name}))`

That way, all media skills could read the `sender` and verify whether the message was sent by itself or not. It can then decide to process it or not.